internal/dcloud/resource_inbound_proxy.go

```diff
@@ -0,0 +1,144 @@
+package dcloud
+
+import (
+	"context"
+	"github.com/cisco-open/dcloud-tb-go-client/tbclient"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceInboundProxy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceInboundProxyCreate,
+		ReadContext:   resourceInboundProxyRead,
+		DeleteContext: resourceInboundProxyDelete,
+		Schema: map[string]*schema.Schema{
+			"uid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"topology_uid": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"nic_uid": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"nic_ip_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"target_vm_name": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
I think this should be computed

Reply:
I get a 400 Bad Request if i omit Vm Name. 
<img width="1155" alt="200-Suc" src="https://github.com/cisco-open/terraform-provider-dcloud/assets/25028473/40636bc7-673a-4a2d-aa0f-231453e899e7">
<img width="1095" alt="400-BR" src="https://github.com/cisco-open/terraform-provider-dcloud/assets/25028473/161c018b-1f8e-4077-b433-c841ccc740ed">

Reply:
Thanks, that's unexpected.  This is one of a handful of cases where we have to make life a little easier for our users, let's just hardcode an empty string in the "Create" method for VM Name, and that way we can mark the value as computed, and still submit a value to the API

Reply:
Thanks Hugh. Made the above stated changes in the latest commit.